### PR TITLE
Add simple sample for service invocation

### PR DESCRIPTION
**Pull request by mukundansundar**
# Description

Add a sample which performs stateless service invocation which shows a simple example of how Dapr can be used with any statestore or docker dependency.

## Issue reference

We strive to have all PRs being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: #247 

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [X] The sample code compiles correctly
* [X] You've tested new builds of the sample if you changed sample code
* [X] You've updated the sample's README if necessary

### CODE REVIEW

File: 11.hello-dapr-slim/README.md [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+Follow [instructions](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md#environment-setup) to download and install the Dapr CLI and initialize Dapr.
+
+## Step 2 - Understand the Code
+
+Now that we've locally set up Dapr, clone the repo, then navigate to the Hello World sample: 
+
+```bash
+git clone https://github.com/dapr/samples.git
+cd samples/1.hello-world
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:22]: please change this to 11.hello-dapr-slim

File: 11.hello-dapr-slim/README.md [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+![Architecture Diagram](./img/Architecture_DiagramA.png)
+
+## Prerequisites
+This sample requires you to have the following installed on your machine:
+- [Node.js version 8 or greater](https://nodejs.org/en/) 
+- [Postman](https://www.getpostman.com/) [Optional]
+
+## Step 1 - Setup Dapr 
+
+Follow [instructions](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md#environment-setup) to download and install the Dapr CLI and initialize Dapr.
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:14]: Docker is mentioned as a prerequisite in the instructions linked here, can we provide instructions to use the --slim flag?

Comment (mukundansundar) [reply 11.hello-dapr-slim/README.md:14]: modified in the docs repo PR

**Comment by LMWF** [comment]
Should this actually be "sample 1"?  The number should be lower since simple or "slim" setups are typically first.

File: 11.hello-dapr-slim/README.md [hunk 3]
```diff
@@ -0,0 +1,148 @@
+# Hello World(without docker)
+
+This tutorial will demonstrate how to get Dapr running locally on your machine without docker. We'll be deploying a Node.js app that subscribes to order messages demostrating service invocation capability of Dapr. The following architecture diagram illustrates the components that make up the first part sample:
```
Comment (LMWF) [inline 11.hello-dapr-slim/README.md:3]: Avoid "we" - https://github.com/dapr/docs/tree/master/contributing

File: 11.hello-dapr-slim/README.md [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+== APP == Got a new order! Order ID: 42
+== APP == Successfully persisted state.
+```
+
+## Step 5 - Confirm Successful Service Invocation
+
+Now, let's just make sure that our method definition was successfully invoked. The response should be 
+
+```
+Got a new order! Oredr ID: 41
```
Comment (LMWF) [inline 11.hello-dapr-slim/README.md:125]: Typo

File: 11.hello-dapr-slim/README.md [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+The command should output text that looks like the following, along with logs:
+
+```
+Starting Dapr with id nodeapp. HTTP Port: 3500. gRPC Port: 9165
+You're up and running! Both Dapr and your app logs will appear here.
+...
+```
+> **Note**: the `--app-port` (the port the app runs on) is configurable. Our Node app happens to run on port 3000, but we could configure it to run on any other port. Also note that the Dapr `--port` parameter is optional, and if not supplied, a random available port is used.
+
+The `dapr run` command looks for the default components directory which for Linux/MacOS is `$HOME/.dapr/components` and for Windows is `%USERPROFILE%\.dapr\components`. On initializing in self hosted mode without docker, the components folder is empty.
```
Comment (LMWF) [inline 11.hello-dapr-slim/README.md:66]: Should this paragraph be omitted?  They wouldn't be using any component in slim mode.

**Comment by mukundansundar** [comment]
> Should this actually be "sample 1"? The number should be lower since simple or "slim" setups are typically first.

Hi Leon, Both the hello-world and the hello-kubernetes samples are straightforward and easy to use and modify. This sample though is a bit more involved in the setup and use, that is why I put it as sample 11.

File: 11.hello-dapr-slim/README.md [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+- [Node.js version 8 or greater](https://nodejs.org/en/) 
+- [Postman](https://www.getpostman.com/) [Optional]
+
+## Step 1 - Setup Dapr 
+
+Follow [instructions](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md#environment-setup) to download and install the Dapr CLI and initialize Dapr.
+
+## Step 2 - Understand the Code
+
+Now that we've locally set up Dapr, clone the repo, then navigate to the Hello World sample:
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:18]: Change reference to "we"

File: 11.hello-dapr-slim/README.md [hunk 7]
```diff
@@ -0,0 +1,148 @@
...
+```js
+app.post('/neworder', bodyParser.json(), (req, res) => {
+    const data = req.body.data;
+    const orderId = data.orderId;
+    console.log("Got a new order! Order ID: " + orderId);
+    res.status(200).send("Got a new order! Order ID: " + orderId);
+});
+```
+
+Here we're exposing an endpoint that will receive and handle `neworder` messages. We log the incoming message, and respond with the same log.
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:38]: Remove "we"

File: 11.hello-dapr-slim/README.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
...
+```
+Starting Dapr with id nodeapp. HTTP Port: 3500. gRPC Port: 9165
+You're up and running! Both Dapr and your app logs will appear here.
+...
+```
+> **Note**: the `--app-port` (the port the app runs on) is configurable. Our Node app happens to run on port 3000, but we could configure it to run on any other port. Also note that the Dapr `--port` parameter is optional, and if not supplied, a random available port is used.
+
+## Step 4 - Post Messages to your Service
+
+Now that Dapr and our Node.js app are running, let's POST messages against it, using different tools. **Note**: here we're POSTing against port 3500 - if you used a different port, be sure to update your URL accordingly.
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:67]: Remove reference to "our" and "we"

File: 11.hello-dapr-slim/README.md [hunk 9]
```diff
@@ -0,0 +1,148 @@
...
+```sh
+dapr invoke --app-id nodeapp --method neworder --payload '{\"data\": { \"orderId\": \"41\" } }'
+```
+
+Linux or MacOS
+```sh
+dapr invoke --app-id nodeapp --method neworder --payload '{"data": { "orderId": "41" } }'
+```
+
+Now, we can also do this using `curl` with:
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:86]: "we"

File: 11.hello-dapr-slim/README.md [hunk 10]
```diff
@@ -0,0 +1,148 @@
...
+
+```sh
+curl -XPOST -d @sample.json -H "Content-Type:application/json" http://localhost:3500/v1.0/invoke/nodeapp/method/neworder
+```
+
+Or, we can also do this using the Visual Studio Code [Rest Client Plugin](https://marketplace.visualstudio.com/items?itemName=humao.rest-client)
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:92]: "we"

File: 11.hello-dapr-slim/README.md [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+POST http://localhost:3500/v1.0/invoke/nodeapp/method/neworder
+
+{
+  "data": {
+    "orderId": "42"
+  } 
+}
+```
+
+Last but not least, we can use the Postman GUI.
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:105]: "Last but not least" is not the tone we usually have in the docs, please change. Also reference to "we"

File: 11.hello-dapr-slim/README.md [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+![Postman Screenshot](./img/postman1.jpg)
+In your terminal window, you should see logs indicating that the message was received and state was updated:
+```bash
+== APP == Got a new order! Order ID: 42
+== APP == Successfully persisted state.
+```
+
+## Step 5 - Confirm Successful Service Invocation
+
+Now, let's just make sure that our method definition was successfully invoked. The response should be
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:117]: Remove first person "lets"

**Comment by orizohar** [comment]
> > Should this actually be "sample 1"? The number should be lower since simple or "slim" setups are typically first.
> 
> Hi Leon, Both the hello-world and the hello-kubernetes samples are straightforward and easy to use and modify. This sample though is a bit more involved in the setup and use, that is why I put it as sample 11.

We'll be removing the numbering from samples in the near future so for now it is fine

File: 11.hello-dapr-slim/README.md [hunk 13]
```diff
@@ -9,13 +9,13 @@ This sample requires you to have the following installed on your machine:
 - [Node.js version 8 or greater](https://nodejs.org/en/) 
 - [Postman](https://www.getpostman.com/) [Optional]
 
-## Step 1 - Setup Dapr 
+## Step 1 - Setup Dapr (Minimal Init)
```
Comment (orizohar) [inline 11.hello-dapr-slim/README.md:12]: Lets be consistent and and call it "slim init" instead of minimal

File: 11.hello-dapr-slim/README.md [hunk 14]
```diff
@@ -0,0 +1,148 @@
+# Hello World (Slim Init, without docker dependency)
+
+This tutorial will demonstrate how to get Dapr running locally on your machine without docker. You will be deploying a Node.js app that subscribes to order messages demostrating service invocation capability of Dapr. The following architecture diagram illustrates the components that make up this sample:
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:3]: demostrating -> demonstrating

File: 11.hello-dapr-slim/README.md [hunk 15]
```diff
@@ -0,0 +1,148 @@
...
+![Architecture Diagram](./img/Architecture_DiagramA.png)
+
+## Prerequisites
+This sample requires you to have the following installed on your machine:
+- [Node.js version 8 or greater](https://nodejs.org/en/) 
+- [Postman](https://www.getpostman.com/) [Optional]
+
+## Step 1 - Setup Dapr (Slim Init)
+
+Follow [instructions](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md#environment-setup) to download and install the Dapr CLI and initialize Dapr.
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:14]: now that we have instructions for self-hosted mode can we use it instead? Or we can change this later once docs PR is merged.

..../environment-setup.md#environment-setup ---> /environment-setup.md#installing-dapr-in-self-hosted-mode

Comment (mukundansundar) [reply 11.hello-dapr-slim/README.md:14]: i can create a separate PR for doc updates if needed.

File: 11.hello-dapr-slim/README.md [hunk 16]
```diff
@@ -0,0 +1,148 @@
...
+
+## Step 2 - Understand the Code
+
+Now that Dapr has been setup locally, clone the repo, then navigate to the Hello World sample:
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:18]: Hello World sample -> hello-dapr-slim sample

File: 11.hello-dapr-slim/README.md [hunk 17]
```diff
@@ -0,0 +1,148 @@
...
+
+```bash
+dapr stop --app-id nodeapp
+```
+
+To see that services have stopped running, run `dapr list`, noting that your services no longer appears!
+
+## Slim Init vs Default Init
+
+With the run of `dapr init --slim`, only the binaries `daprd` and `placement` are instaled in the system. In the normal default init, `dapr init` placement service is installed as a container and additional redis and zipkin containers are run for enabling state store and tracing, which requires Docker to be installed as a prerequisite. See [environment setup](https://github.com/dapr/docs/blob/master/getting-started/environment-setup.md) for more information. The slim init provides only limited functionaity unless you [configure](https://github.com/dapr/docs/tree/master/howto/configure-redis#Self-Hosted-Mode-without-Containers) a state store yourself.
```
Comment (pruthvidhodda) [inline 11.hello-dapr-slim/README.md:141]: instaled --> installed

**Pull request merged**